master/static/migrations/20231126215150_add-checkpoints-trigger.tx.up.sql

```diff
@@ -0,0 +1,177 @@
+ALTER TABLE checkpoints_v2 ADD COLUMN IF NOT EXISTS seq bigint;
+
+-- the sequence
+CREATE SEQUENCE IF NOT EXISTS stream_checkpoint_seq START 1;
+
+-- trigger function to update sequence number on row modification
+-- this should fire BEFORE so that it can modify NEW directly.
+CREATE OR REPLACE FUNCTION stream_checkpoint_seq_modify() RETURNS TRIGGER AS $$
+BEGIN
+    NEW.seq = nextval('stream_checkpoint_seq');
+RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+--
+DROP TRIGGER IF EXISTS stream_checkpoint_trigger_seq ON checkpoints_v2;
+CREATE TRIGGER stream_checkpoint_trigger_seq
+    BEFORE INSERT OR UPDATE OF
+    task_id, allocation_id, report_time, state, resources, metadata, size
+                     ON checkpoints_v2
+                         FOR EACH ROW EXECUTE PROCEDURE stream_checkpoint_seq_modify();
+
+-- helper function to create a checkpoint jsonb object for streaming
+CREATE OR REPLACE FUNCTION stream_checkpoint_notify(
+    before jsonb, beforework integer, after jsonb, afterwork integer, trialid integer, expid integer
+) RETURNS integer AS $$
+DECLARE
+output jsonb = NULL;
+    temp jsonb = NULL;
+    temptrial jsonb = NULL;
+    tempexp jsonb = NULL;
+BEGIN
+    IF before IS NOT NULL THEN
+        temp = before || jsonb_object_agg('workspace_id', beforework);
+        temptrial = temp || jsonb_build_object('trial_id', trialid);
+        tempexp = temp || jsonb_build_object('experiment_id', expid);
```

Comment:
nit: it doesn't look like `temptrial` is actually doing anything did you intend for it to be:
```suggestion
        tempexp = temptrial || jsonb_build_object('experiment_id', expid);
```